server/infinispan/src/main/java/org/hawkular/apm/server/infinispan/InfinispanPropertyCache.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.server.infinispan;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.Resource;
+
+import org.hawkular.apm.api.services.ServiceLifecycle;
+import org.hawkular.apm.api.utils.PropertyUtil;
+import org.hawkular.apm.server.api.services.CacheException;
+import org.hawkular.apm.server.api.services.Property;
+import org.hawkular.apm.server.api.services.PropertyService;
+import org.infinispan.Cache;
+import org.infinispan.manager.CacheContainer;
+import org.infinispan.notifications.Listener;
+import org.infinispan.notifications.cachelistener.annotation.CacheEntryModified;
+import org.infinispan.notifications.cachelistener.event.CacheEntryModifiedEvent;
+import org.infinispan.notifications.cachelistener.event.impl.EventImpl;
+
+/**
+ * @author Pavol Loffay
+ */
+public class InfinispanPropertyCache implements PropertyService, ServiceLifecycle {
+
+    private static final Logger log = Logger.getLogger(InfinispanPropertyCache.class.getName());
+
+    protected static final String CACHE_NAME = "property";
+
+    @Resource(lookup = "java:jboss/infinispan/APM")
+    private CacheContainer cacheContainer;
+
+    private Cache<String, Property<?>> propertyCache;
+    private CacheEventListener eventListener;
+
+
+    public InfinispanPropertyCache() {
+        this.eventListener = new CacheEventListener();
+    }
+
+    public InfinispanPropertyCache(Cache<String, Property<?>> cache) {
+        this();
+        this.propertyCache = cache;
+        this.propertyCache.addListener(eventListener);
+    }
+
+    @Override
+    @PostConstruct
+    public void init() {
+        // If cache container not already provisions, then must be running outside of a JEE
+        // environment, so create a default cache container
+        if (cacheContainer == null) {
```

Comment:
May want to have a guard around this method, so the init method does nothing if propertyCache already initialised. May only be an issue in tests, but would avoid value provided in constructor being overwritten by mistake.